modules/ds18b20/ds18b20.h

```diff
@@ -0,0 +1,315 @@
+ * @email   tilen@majerle.eu
+ * @website http://stm32f4-discovery.net
+ * @link    http://stm32f4-discovery.net/2015/07/hal-library-06-ds18b20-for-stm32fxxx/
+ * @version v1.0
+ * @ide     Keil uVision
+ * @license MIT
+ * @brief   Library for interfacing DS18B20 temperature sensor from Dallas semiconductors.
+ *	
+\verbatim
+   ----------------------------------------------------------------------
+    Copyright (c) 2016 Tilen MAJERLE
+
+    Permission is hereby granted, free of charge, to any person
+    obtaining a copy of this software and associated documentation
+    files (the "Software"), to deal in the Software without restriction,
+    including without limitation the rights to use, copy, modify, merge,
+    publish, distribute, sublicense, and/or sell copies of the Software, 
+    and to permit persons to whom the Software is furnished to do so, 
+    subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be
+    included in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+    EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+    OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE
+    AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+    HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+    WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING 
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+    OTHER DEALINGS IN THE SOFTWARE.
+   ----------------------------------------------------------------------
+\endverbatim
+ */
+#ifndef DS18B20_H
+#define DS18B20_H
+
+/**
+ * @addtogroup STM32Fxxx_HAL_Libraries
+ * @{
+ */
+
+/**
+ * @defgroup DS12820
+ * @brief    Library for interfacing DS18B20 temperature sensor from Dallas semiconductors - http://stm32f4-discovery.net/2015/07/hal-library-06-ds18b20-for-stm32fxxx/
+ * @{
+ *
+ * With this you can read temperature, set and get temperature resolution from 9 to 12 bits and check if device is DS18B20.
+ * 
+ * Pin for STM32Fxxx is the same as set with @ref OneWireStruct library.
+ *
+ * \par Changelog
+ *
+\verbatim
+ Version 1.0
+  - First release
+\endverbatim
+ *
+ * \par Dependencies
+ *
+\verbatim
+ - STM32Fxxx HAL
+ - TM OneWireStruct
+ - TM GPIO
+ - defines.h
+\endverbatim
+ */
+
+#include <modules/onewire/onewire.h>
+
+/* OneWireStruct version check */
+// #if ONEWIRE_H < 100
+// #error "Please update TM OneWireStruct LIB, minimum required version is 2.0.0. Download available on stm32f4-discovery.net website"
+// #endif
+
+/**
+ * @defgroup DS18B20_Macros
+ * @brief    Library defines
+ * @{
+ */
+
+/* More details: https://datasheets.maximintegrated.com/en/ds/DS18B20.pdf */
+
+/* Every OneWireStruct chip has different ROM code, but all the same chips has same family code */
+/* in case of DS18B20 this is 0x28 and this is first byte of ROM address */
+#define DS18B20_FAMILY_CODE				0x28

```

Comment:
Since this is now "our" code rather than just something "imported", our coding standard state the use of tabs over spaces.  So please replace tabs with spaces.